### PR TITLE
Attendance check created after shift permission approved

### PR DESCRIPTION
**Pull request by yusuffGoodie**
## Is this a Feature, Chore or Bug?
- [] Feature
- [] Chore
- [] Bug

## Analysis and design (optional)
Analyse and attach the design documentation

## Solution description
Describe your code changes in detail for reviewers.

## Is there a business logic within a doctype?
    - [] Yes
    - [] No

## Output screenshots (optional)
Post the output screenshots, if a UI is affected or added due to this feature.

## Areas affected and ensured
List out the areas affected by your code changes.

## Is there any existing behavior change of other features due to this code change?
Mention Yes or No. If Yes, provide the appropriate explanation.

## Did you test with the following dataset?
- [] Existing Data
- [] New Data

## Was child table created?
    - [] is attachment required?
        did you test attachment
## Did you delete custom field?
    - [] Yes
    - [] No
        If yes, did you write a delete patch?

## Is patch required?
- [] Yes
- [] No
    ## Was the patch test?

## Which browser(s) did you use for testing?
  - [] Chrome
  - [] Safari
  - [] Firefox

### CODE REVIEW

File: one_fm/hiring/utils.py [hunk 1]
```diff
@@ -684,7 +684,7 @@ def get_interview_skill_and_question_set(interview_round, interviewer=False, int
 		interview_feedback = frappe.get_doc('Interview Feedback', feedback_exists)
 		return interview_feedback.interview_question_assessment, interview_feedback.skill_assessment, feedback_exists
 	else:
-		question = frappe.get_all('Interview Questions', filters ={'parent': interview_round}, fields=['questions', 'answer', 'weight'])
+		question = frappe.get_all('Interview Questions', filters ={'parent': interview_round}, fields=['questions', 'answer', 'weight'], order_by='idx')
```
Comment (NoBoneZ) [inline one_fm/hiring/utils.py]: Please, why are there changes relating to interview here

File: one_fm/public/js/doctype_js/interview.js [hunk 2]
```diff
@@ -130,7 +129,7 @@ frappe.ui.form.on('Interview', {
 			fieldname: 'weight',
 			label: __('Weight'),
 		}, {
-			fieldtype: 'Data',
+			fieldtype: 'Small Text',
```
Comment (NoBoneZ) [inline one_fm/public/js/doctype_js/interview.js]: Please, why is this change here

File: one_fm/one_fm/doctype/attendance_check/attendance_check.py [hunk 3]
```diff
@@ -301,10 +301,16 @@ def create_attendance_check(attendance_date=None):
         if not attendance_date:
             attendance_date = add_days(today(), -1)
 
+        employee_with_shift_permission = [i.employee for i in frappe.db.get_list('Shift Permission', {
```
Comment (NoBoneZ) [inline one_fm/one_fm/doctype/attendance_check/attendance_check.py]: you could use the frappe.db.get_list("Shift Permission", filter=filter, pluck="employee), please check this

File: one_fm/one_fm/doctype/attendance_check/attendance_check.py [hunk 4]
```diff
@@ -301,10 +301,16 @@ def create_attendance_check(attendance_date=None):
...
+            'date': attendance_date
+        }
+        )]
+
         absentees = frappe.get_all("Attendance", filters={
             'docstatus':1,
             'status':'Absent',
-            'attendance_date':attendance_date},
+            'attendance_date':attendance_date,
+            'employee':["not in", employee_with_shift_permission]},
```
Comment (NoBoneZ) [inline one_fm/one_fm/doctype/attendance_check/attendance_check.py:313]: so instead of declaring it as a variable you could just writhe the query here, unless you are using it somewhere else

File: one_fm/patches/v15_0/update_job_offer_00057_erf.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import frappe
```
Comment (NoBoneZ) [inline one_fm/patches/v15_0/update_job_offer_00057_erf.py]: please, Did you check out from version-15 ?

File: one_fm/one_fm/doctype/attendance_check/attendance_check.py (hunk 3 above)
Comment (yusuffGoodie) [reply one_fm/one_fm/doctype/attendance_check/attendance_check.py]: Noted

File: one_fm/hiring/utils.py (hunk 1 above)
Comment (yusuffGoodie) [reply one_fm/hiring/utils.py]: I didn't touch this file

File: one_fm/one_fm/doctype/attendance_check/attendance_check.py (hunk 4 above)
Comment (yusuffGoodie) [reply one_fm/one_fm/doctype/attendance_check/attendance_check.py:313]: I used variable to make code clean and clearer for someone else. If I add everything in a single it may look ambiguous for new developer

File: one_fm/patches/v15_0/update_job_offer_00057_erf.py (hunk 5 above)
Comment (yusuffGoodie) [reply one_fm/patches/v15_0/update_job_offer_00057_erf.py]: I didn't touch this either

File: one_fm/public/js/doctype_js/interview.js (hunk 2 above)
Comment (yusuffGoodie) [reply one_fm/public/js/doctype_js/interview.js]: I didn't touch this too

**Review by Talleyrand333 — APPROVED** [review]
